dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx

```diff
@@ -1154,6 +1160,103 @@ export function JsonRpcContextProvider({
     ),
   };
 
+  // -------- TEZOS RPC METHODS --------
+
+  const tezosRpc = {
+    testGetAccounts: _createJsonRpcRequestHandler(
+      async (
+        chainId: string,
+        address: string
+      ): Promise<IFormattedRpcResponse> => {
+        try {
+          const result = await client!.request<{ signature: string }>({
+            chainId,
+            topic: session!.topic,
+            request: {
+              method: DEFAULT_TEZOS_METHODS.TEZOS_GET_ACCOUNTS,
+              params: {},
+            },
+          });
+
+          return {
+            method: DEFAULT_TEZOS_METHODS.TEZOS_GET_ACCOUNTS,
+            address,
+            valid: true,
+            result: JSON.stringify(result, null, 2),
+          };
+        } catch (error: any) {
+          throw new Error(error.message);
+        }
+      }
+    ),
+    testSignTransaction: _createJsonRpcRequestHandler(
+      async (
+        chainId: string,
+        address: string
+      ): Promise<IFormattedRpcResponse> => {
+        try {
+          const result = await client!.request<{ hash: string }>({
+            chainId,
+            topic: session!.topic,
+            request: {
+              method: DEFAULT_TEZOS_METHODS.TEZOS_SEND,
+              params: {
+                account: address,
+                operations: [
+                  {
+                    kind: "transaction",
+                    amount: "1", // 1 mutez, smallest unit
+                    destination: address, // send to ourselves
+                  },
+                ],
+              },
+            },
+          });
+
+          return {
+            method: DEFAULT_TEZOS_METHODS.TEZOS_SEND,
+            address,
+            valid: true,
+            result: result.hash,
+          };
+        } catch (error: any) {
+          throw new Error(error.message);
+        }
+      }
+    ),
+    testSignMessage: _createJsonRpcRequestHandler(
+      async (
+        chainId: string,
+        address: string
+      ): Promise<IFormattedRpcResponse> => {
+        const payload = "05010000004254";
+
+        try {
+          const result = await client!.request<{ signature: string }>({
+            chainId,
+            topic: session!.topic,
+            request: {
+              method: DEFAULT_TEZOS_METHODS.TEZOS_SIGN,
+              params: {
+                account: address,
+                payload,
+              },
+            },
+          });
+
+          return {
+            method: DEFAULT_TEZOS_METHODS.TEZOS_SIGN,
+            address,
+            valid: true,
```

Comment:
Is there a way we can validate the signature here to evaluate `valid` properly?

Reply:
It should be possible if we have access to the public key. But to have the public key, `tezos_getAccounts` has to be called before (or we have to fetch the public key from chain, which is not always possible). So it may not always be possible to do the validation. Should I still add it and it may not be possible sometimes, or just remove the "valid" flag?

Reply:
I see, would be great to have proper validation if you think it can be done here but not a showstopper for now 👍

Reply:
We can do this in the future (it's not as straight forward because of the reasons mentioned above), but as far as I can see all other protocols also have `valid: true` hardcoded as well.